### PR TITLE
Refactored multiple assertion tests into parameterized ones

### PR DESCRIPTION
**Pull request by eas5**
Problem:
A test method with many individual assertions stops being executed on the first failed assertion, which prevents the remaining ones' execution. In the refactored methods, the difference between the assertions lies in different arguments only.

Solution:
Parameterized tests make it possible to run a test multiple times, with different arguments, as individual and independent tests. This way, we were able to make 2 original tests become 11 independent ones. In this refactoring, no original assertion parameter was changed.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

N/A

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]

**Review by basil — APPROVED** [review]

**Comment by MarkEWaite** [comment]
This PR is now ready for merge.  We will merge it after approximately 24 hours if there is no negative feedback.

**Review by oleg-nenashev — APPROVED** [review]
👍

File: cli/src/test/java/hudson/cli/HexDumpTest.java
Comment (MarkEWaite) [inline cli/src/test/java/hudson/cli/HexDumpTest.java:17]: Optionally, you might consider including the expected value in the test name for easier diagnosis in case of failure.
```suggestion
  @ParameterizedTest(name = "expected: {0}")
```

File: cli/src/test/java/hudson/cli/HexDumpTest.java
Comment (MarkEWaite) [inline cli/src/test/java/hudson/cli/HexDumpTest.java:33]: Optionally, you might consider including the expected value and the start and end values in the test name for easier diagnosis in case of failure.
```suggestion
  @ParameterizedTest(name = "expected: {0}, start: {3}, end: {4}")
```

**Review by MarkEWaite — APPROVED** [review]
I confirmed that the tests are running as expected in the CI job.  While reading the test results, I saw that it might be a little easier to read the test results if the test output included one or more of the test parameters.

Proposed change is optional and does not block the merge.  Whether the proposed changes are applied or not, I approve this change.  Thanks for contributing!

**Review by StefanSpieker — APPROVED** [review]

File: cli/src/test/java/hudson/cli/HexDumpTest.java
Comment (eas5) [reply cli/src/test/java/hudson/cli/HexDumpTest.java]: Hi @MarkEWaite, I have done your suggestions. Test names are now generated including expected values and passed parameters, as suggested. Thank you!

**Pull request merged**
